### PR TITLE
MAGECLOUD-4606: Review Hotfixes List and Add All Missing for 2.1.4+

### PR DESCRIPTION
**Pull request by NadiyaS**
### Description
Check all patches from https://magento.com/tech-resources/download for 2.1.4+ versions and bundle missing patches to magento-cloud-patches

### Manual testing scenarios
check edge versions to which patches should be applied
e.g. if patch should be applied for >=2.1.4 <2.2.0, need to check 2.1.4 version and on the latest 2.1 version - 2.1.18.
There is one custom patch - PRODSECBUG-2233  for 2.2.0-2.2.6 versions. It includes fix for PRODSECBUG-2233 and part of fix from https://jira.corp.magento.com/browse/MAGETWO-95236. So, we need to test functionality it fixes on one of 2.2.0-2.2.6 version. Description of it can be taken from https://jira.corp.magento.com/browse/PRODSECBUG-2233

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: patches.json [hunk 1]
```diff
@@ -205,6 +205,39 @@
...
+    "Pass Store View scope in the Async/Bulk Web API": {
+      "2.3.1": "MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.1.patch",
+      "2.3.2": "MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.2.patch"
+    },
+    "Admin path disclosure bug": {
+      ">=2.1.4 <2.2.0": "PRODSECBUG-2432__admin_path_disclosure_bug__2.1.4.patch",
+      ">=2.2.0 <2.3.0": "PRODSECBUG-2432__admin_path_disclosure_bug__2.2.0.patch",
+      "2.3.0 - 2.3.2": "PRODSECBUG-2432__admin_path_disclosure_bug__2.3.0.patch"
+    },
+    "Pre-auth SQL Injection": {
```
Comment (YPyltiai) [inline patches.json]: I think it's same as: https://github.com/magento/magento-cloud-patches/blob/develop/patches.json#L164

File: patches/MAGETWO-92926__fix_for_multi-site_configuration_issue__2.2.4.patch [hunk 2]
```diff
@@ -0,0 +1,16 @@
+diff --git a/app/etc/di.xml b/app/etc/di.xml
```
Comment (YPyltiai) [inline patches/MAGETWO-92926__fix_for_multi-site_configuration_issue__2.2.4.patch]: Change first line to be same as our other patches with parameters.

File: patches/MAGETWO-93036__cannot_change_the_applied_theme__2.2.5.patch [hunk 3]
```diff
@@ -0,0 +1,17 @@
+diff --git a/vendor/magento/module-email/Model/AbstractTemplate.php b/vendor/magento/module-email/Model/AbstractTemplate.php
+index fa9d28074bf8..81e993fad76d 100644
+--- a/vendor/magento/module-email/Model/AbstractTemplate.php
++++ b/vendor/magento/module-email/Model/AbstractTemplate.php
```
Comment (YPyltiai) [inline patches/MAGETWO-93036__cannot_change_the_applied_theme__2.2.5.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/MAGETWO-92926__fix_for_multi-site_configuration_issue__2.2.4.patch [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+index de50840..4eaa739 100644
```
Comment (YPyltiai) [inline patches/MAGETWO-92926__fix_for_multi-site_configuration_issue__2.2.4.patch]: We were deleting index everywhere. Please drop it.

File: patches/MAGETWO-93083__optimize_retrieving_product_attributes__2.2.5.patch [hunk 5]
```diff
@@ -0,0 +1,185 @@
+diff --git a/vendor/magento/module-catalog/Model/Product.php b/vendor/magento/module-catalog/Model/Product.php
```
Comment (YPyltiai) [inline patches/MAGETWO-93083__optimize_retrieving_product_attributes__2.2.5.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/MAGETWO-95591__customer_attributes_issue__2.2.6.patch [hunk 6]
```diff
@@ -0,0 +1,1592 @@
+diff --git a/vendor/magento/module-catalog/Controller/Adminhtml/Product/Attribute/Save.php b/vendor/magento/module-catalog/Controller/Adminhtml/Product/Attribute/Save.php
```
Comment (YPyltiai) [inline patches/MAGETWO-95591__customer_attributes_issue__2.2.6.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/MAGETWO-98129__transaction_field_is_removed_by_authorize_net__2.2.0.patch [hunk 7]
```diff
@@ -0,0 +1,337 @@
+Index: vendor/magento/module-authorizenet/Model/Directpost.php
```
Comment (YPyltiai) [inline patches/MAGETWO-98129__transaction_field_is_removed_by_authorize_net__2.2.0.patch]: Format of this patch looks odd. Need to re-work it

File: patches/MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.1.patch [hunk 8]
```diff
@@ -0,0 +1,934 @@
+diff --git a/vendor/magento/module-amqp/Plugin/AsynchronousOperations/MassConsumerEnvelopeCallback.php b/vendor/magento/module-amqp/Plugin/AsynchronousOperations/MassConsumerEnvelopeCallback.php
```
Comment (YPyltiai) [inline patches/MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.1.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.2.patch [hunk 9]
```diff
@@ -0,0 +1,948 @@
+diff --git a/vendor/magento/module-amqp/Plugin/AsynchronousOperations/MassConsumerEnvelopeCallback.php b/vendor/magento/module-amqp/Plugin/AsynchronousOperations/MassConsumerEnvelopeCallback.php
```
Comment (YPyltiai) [inline patches/MAGETWO-99902__pass_store_view_scope_in async_web_api__2.3.2.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/PRODSECBUG-2198__preauth_sql__2.1.4.patch [hunk 10]
```diff
@@ -0,0 +1,16 @@
+Index: vendor/magento/framework/DB/Adapter/Pdo/Mysql.php
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2198__preauth_sql__2.1.4.patch]: Format looks odd.

File: patches/PRODSECBUG-2198__preauth_sql__2.2.0.patch [hunk 11]
```diff
@@ -0,0 +1,96 @@
+diff --git a/vendor/magento/module-catalog/Model/Product/ProductFrontendAction/Synchronizer.php b/vendor/magento/module-catalog/Model/Product/ProductFrontendAction/Synchronizer.php
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2198__preauth_sql__2.2.0.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/PRODSECBUG-2198__preauth_sql__2.3.0.patch [hunk 12]
```diff
@@ -0,0 +1,312 @@
+diff --git a/vendor/magento/module-catalog/Model/Product/ProductFrontendAction/Synchronizer.php b/vendor/magento/module-catalog/Model/Product/ProductFrontendAction/Synchronizer.php
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2198__preauth_sql__2.3.0.patch]: Change first line to be same as our other patches with parameters. Also drop index

File: patches/PRODSECBUG-2233__fix_xss_in_order_history__2.2.0.patch [hunk 13]
```diff
@@ -0,0 +1,123 @@
+Index: vendor/magento/module-sales/Helper/Admin.php
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2233__fix_xss_in_order_history__2.2.0.patch]: Format is not as we use again.

File: patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.1.4.patch [hunk 14]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.1.4.patch]: Format + Change first line to be same as our other patches with parameters. Also drop index

File: patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.2.0.patch [hunk 15]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.2.0.patch]: Format + Change first line to be same as our other patches with parameters. Also drop index

File: patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.3.0.patch [hunk 16]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (YPyltiai) [inline patches/PRODSECBUG-2432__admin_path_disclosure_bug__2.3.0.patch]: Format + Change first line to be same as our other patches with parameters. Also drop index

File: patches.json (hunk 1 above)
Comment (NadiyaS) [reply patches.json]: yes, you are right. changes were in doc blocks only

**Review by oshmyheliuk — DISMISSED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
Waiting for environment fix. Ticket  in zendesk 264557

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**